aptos-move/aptos-vm-types/src/module_and_script_storage/state_view_adapter.rs

```diff
@@ -72,6 +82,62 @@ impl<'s, S: StateView, E: WithRuntimeEnvironment> AptosCodeStorageAdapter<'s, S,
         let storage = adapter.into_unsync_code_storage(runtime_environment);
         Self { storage }
     }
+
+    /// Drains cached verified modules from the code storage, transforming them into format used by
+    /// global caches. Should only be called when the code storage borrows [StateView].
```

Comment:
can we briefly explain why?

Reply:
The problem here was that we have ownership of state view or it's reference, but I fixed that - we just need to unpack the state view together with module cache verified modules iterator